File: Data Structures & Algorithms/anagram-groups/submission-0.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        //first loop through all strings
        unordered_map<string, vector<string>> m;
        vector<vector<string>> ans;
        //this map stores the root anagram as the key, and then stores vector as value
        for (string i : strs){
            string v = i;
            sort(v.begin(), v.end());
            m[v].push_back(i);
        }

        for (auto vec : m){
            ans.push_back(vec.second);
        }

        return ans;
    }
};

File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        final = ""
        for j in range(len(strs[0])):
            c = ""
            for i in range(len(strs)):
                if j < len(strs[i]):
                    if not c:
                        c = strs[i][j]
                    elif strs[i][j]!=c:
                        return final
                    else:
                        continue
                else:
                    return final
            final+=c

        return final